### ISSUE TITLE
Migrate user and entity models and services to use Prisma

### ISSUE THREAD

**Issue by sherryhli** [uwblueprint/sistering#14]
## User Story and Context

Following #8, we have decided to migrate from Sequelize to Prisma for our ORM. #12 installed Prisma as a dependency and tested the connection logic, now we need to migrate existing models and services to use Prisma.

### Product
N/A

### Technical
* `User` and `Entity` models are defined using Prisma instead of Sequelize
* `UserService` and `EntityService` use Prisma client methods instead of Sequelize client methods for CRUD operations
* Public APIs over users and entities remain unchanged (test all queries and mutations [here](https://uwblueprint.github.io/starter-code-v2/docs/user-auth/testing.html#graphql) and [here](https://uwblueprint.github.io/starter-code-v2/docs/crud/testing.html#graphql) to ensure their behaviours have not changed)
* All references to Sequelize methods are removed from the code

## Dev Notes

You'll largely be following this official Prisma [guide](https://www.prisma.io/docs/getting-started/setup-prisma/add-to-existing-project/relational-databases-typescript-postgres). Please read it through **and the additional notes below** before starting this task.

While developing, you should be running the application with Docker and running all yarn and npx commands from the backend container:
```
# from the root of the sistering folder
$ docker-compose up --build

# get the backend container name, it should be sistering_backend_1
$ docker ps

# run a bash shell in the backend container
$ docker exec -it <CONTAINER-NAME> /bin/bash
```

Additional Notes:
* Wait for #13 to be merged into main before branching off
* You'll need to add the following line to `backend/prisma/.env`
```
DATABASE_URL="postgresql://${POSTGRES_USER}:${POSTGRES_PASSWORD}@${DB_HOST}:5432/${POSTGRES_DB}"
```
* To create the Prisma models, you can use [introspection](https://www.prisma.io/docs/getting-started/setup-prisma/add-to-existing-project/relational-databases/introspection-typescript-postgres), however, please check the following for the generated models:
    * Sequelize automatically included `createdAt` and `updatedAt` columns for all tables but we don't currently have any use for them so they can be excluded from the Prisma model
    * Following TypeScript/JavaScript convention, we use camelCase field names in the model, but following PostgreSQL convention, we use snake_case for database columns. Similarly, while the model name starts with a capital letter according to convention, the table name should be all lowercase. Ensure that these differences are reflected in the model, you may need to use the `@map` directive, [example](https://github.com/prisma/prisma/discussions/2530):
```
model Post {
  id       Int      @default(autoincrement()) @id
  authorId Int?     @map("author_id")
  isActive Boolean? @default(false) @map("is_active")

  @@map("posts")
}
```
* At this point, prior to running Prisma migrations, you will likely want to comment out or delete the Sequelize initialization logic in server.ts. You'll also want to take down your database and all of its current data with `docker-compose down --volumes`
* Running `docker-compose down --volumes` will delete your test admin user, so to test the GraphQL API, create a new user for yourself by running this insert statement in the DB container
```sql
$ docker exec -it sistering_db_1 /bin/bash -c "psql -U postgres -d sistering"

INSERT INTO users (first_name, last_name, auth_id, role) VALUES ('<FIRST_NAME>', '<LAST_NAME>', '<FIND-YOUR-UID-FROM-FIREBASE>', 'Admin');
```

**Comment by sherryhli** [comment]
Hey @lambo-liu I’m using ZenHub in GitHub, [click this link](https://app.zenhub.com/workspaces/sistering-614f8dad6afeb8001d428787/board) to join my workspace and see other features available in GitHub or [download the ZenHub extension](https://www.zenhub.com/extension) and sign up with your GitHub account. 
  _Posted using [ZenHub](https://www.zenhub.com)_

**Comment by sherryhli** [comment]
Hey @briantu I’m using ZenHub in GitHub, [click this link](https://app.zenhub.com/workspaces/sistering-614f8dad6afeb8001d428787/board) to join my workspace and see other features available in GitHub or [download the ZenHub extension](https://www.zenhub.com/extension) and sign up with your GitHub account. 
  _Posted using [ZenHub](https://www.zenhub.com)_
